5.0/en/0x14-V6-Cryptography.md

```diff
@@ -2,62 +2,318 @@
+| CRC (any length) | --                                                                                                        |
+| MD4              | [RFC 1320](https://www.rfc-editor.org/info/rfc1320)                                                       |
+| MD5              | [RFC 1321](https://www.rfc-editor.org/info/rfc1321)                                                       |
+| SHA-1            | [RFC 3174](https://www.rfc-editor.org/info/rfc3174) & [RFC 6194](https://www.rfc-editor.org/info/rfc6194) |
+
+### Disallowed Hash Functions for Digital Signatures
+
+For digital signature implementations, the following hash functions MUST NOT be used due to insufficient collision resistance:
+
+| Hash functions | Reference                                                      |
+| -------------- | -------------------------------------------------------------- |
+| SHA-224        | [FIPS 180-4](https://csrc.nist.gov/pubs/fips/180-4/upd1/final) |
+| SHA-512/224    | [FIPS 180-4](https://csrc.nist.gov/pubs/fips/180-4/upd1/final) |
+| SHA3-224       | [FIPS 202](https://csrc.nist.gov/pubs/fips/202/final)          |
+
+## V6.4 Random Values
 
 Cryptographically secure Pseudo-random Number Generation (CSPRNG) is incredibly difficult to get right. Generally, good sources of entropy within a system will be quickly depleted if over-used, but sources with less randomness can lead to predictable keys and secrets.
 
-| # | Description | L1 | L2 | L3 | CWE |
-| :---: | :--- | :---: | :---: | :---: | :---: |
-| **6.3.1** | [MODIFIED] Verify that all random numbers, random file names, and random strings are generated using a cryptographically-secure pseudo-random number generator (CSPRNG) when these random values are intended to be not guessable by an attacker. | | ✓ | ✓ | 338 |
-| **6.3.2** | [MODIFIED] Verify that GUIDs are created with an implementation of the GUID v4 algorithm which utilizes a cryptographically-secure pseudo-random number generator (CSPRNG). GUIDs created using other algorithm versions or using insufficiently secure pseudo-random number generators may be predictable. | | ✓ | ✓ | 338 |
-| **6.3.3** | Verify that random numbers are created with proper entropy even when the application is under heavy load, or that the application degrades gracefully in such circumstances. | | | ✓ | 338 |
+|     #     | Description                                                                                                 | L1  | L2  | L3  | CWE |
+| :-------: | :---------------------------------------------------------------------------------------------------------- | :-: | :-: | :-: | :-: |
+| **6.3.1** | All random numbers and strings intended to be non-guessable must be generated using a CSPRNG.               |     |  ✓  |  ✓  | 338 |
+| **6.3.2** | GUIDs must be created using the GUID v4 algorithm with CSPRNG.                                              |     |  ✓  |  ✓  | 338 |
```

Comment:
This one feelds weird to me. Why is this in cryptography? We are talking about UUIDs, right? There are legitimate use cases for other UUIDs versions. I feel this should be clarified.